### ISSUE TITLE
Authentication error when using OAuth Client

### ISSUE THREAD

**Issue by stoffelio** [slakbal/gotowebinar#12]
Not sure GoTo have changed things on their end or the documentation is incorrect.

You talk about creating an App under "My Apps", but this menu option does not exist in the developer back end. Trying to use an OAuth Clients, which is the option I see in the menu, only results in an authentication error.

`Slakbal\Gotowebinar\Exception\GotoException  : Unauthorized - Could not authenticate with the provided credentials.`

**Comment by slakbal** [comment]
I'll take a look. Goto deployed a new developer backend. Once a client app is setup you're supposed to get the credentials: key and secret

**Comment by dankinchen** [comment]
Yes the new location is under OAuth Clients (https://developer.logmeininc.com/clients). BUT this app uses the DIRECT Login. This page specifically says that any NEW OAuth Clients will not be able to use it as it has been deprecated.

https://developer.goto.com/guides/References/05_Direct-Login_migration/

Once you figure out your credentials and authorize your app, you will get "Invalid_client".

**Comment by wize-wiz** [comment]
@slakbal I started a PR to get this done. Anything you wish to change/add/remove, be my guest.

Receiving an authorization code needs to be done manually, the API does not offer any other way to automate this process. Although I'm thinking of trying Dusk to see if I'm wrong 😄

**Comment by stoffelio** [comment]
@slakbal - have you had a chance to look at the PR? Would be great to get this merged.

**Comment by slakbal** [comment]
Hi Everyone,

Sorry about the delay, the 2020 dumpster-fire just burned me out a bit. I'll be reviewing everything on the package starting next week. Will keep you posted.

**Comment by slakbal** [comment]
I just read on the Goto Docs: https://developer.goto.com/guides/References/05_Direct-Login_migration/

`
**Integrations with System Users**
If the username and password that you use in the password grant flow doesn’t represent real end users, the authorization code flow might not be a good fit for your integration. The process for migrating clients like yours will be defined later and you can continue to use the password grant flow until then.
`

Did anyone find some other information on this yet for system to system integration?

**Comment by stoffelio** [comment]
Reading the guide: https://developer.goto.com/guides/HowTos/03_HOW_accessToken/
Sounds to me like the authorization code needs to be created manually once, but after that we can talk to the api using an accesss token, which can be requested using the refresh token, with the latter being valid for 30 days.
Am I missing the issue?
Edit: So just create a new OAutch client and use the new way I guess.
